### PR TITLE
Look into UI enhancement. #140

### LINKED ISSUE #140: Look into UI enhancement.
Especially Listview.
For example,
when a person is deleted, can we have diminishing effects?

### PR DESCRIPTION
**Pull request by yl-coder**
Fixes #140 

image

### CODE REVIEW

**Comment by yl-coder** [comment]
Will refactor first before asking for review.

**Comment by HansNewbie** [comment]
On how to get GitHub avatar from profile page:
http://stackoverflow.com/questions/22932422/get-github-avatar-from-email-or-name

Add `.png` behind profile URL, say github.com/HansNewbie.png. The link above also states you can specify size.

File: src/main/java/address/controller/PersonCardController.java [hunk 1]
```diff
@@ -68,9 +112,43 @@ protected String computeValue() {
                 return person.tagsString();
             }
         });
+        person.isDeletedProperty().addListener(new ChangeListener<Boolean>() {
```
Comment (YijinL) [inline src/main/java/address/controller/PersonCardController.java]: Can use lambda expression
```java
person.isDeletedProperty().addListener((obs, oldV, newV) -> {
    if (newValue) {
        handleDeletedPerson();
    }
});
```

File: src/main/java/address/controller/PersonCardController.java [hunk 2]
```diff
@@ -68,9 +112,43 @@ protected String computeValue() {
...
+            @Override
+            public void changed(ObservableValue<? extends Boolean> observable, Boolean oldValue, Boolean newValue) {
+                if (newValue == true){
+                    handleDeletedPerson();
+                }
+            }
+        });
+        person.githubProfilePicUrlProperty().addListener(new ChangeListener<String>() {
```
Comment (YijinL) [inline src/main/java/address/controller/PersonCardController.java]: Use lambda expressions too, more concise

File: src/main/java/address/controller/PersonOverviewController.java [hunk 3]
```diff
@@ -67,9 +74,12 @@ public void setConnections(MainController mainController, ModelManager modelMana
...
         int selectedIndex = personList.getSelectionModel().getSelectedIndex();
         if (selectedIndex >= 0) {
+            Person tmpPerson = personList.getItems().get(selectedIndex);
             mainController.getStatusBarHeaderController().postStatus(
-                                                    new PersonDeletedStatus(personList.getItems().get(selectedIndex)));
-            modelManager.deletePerson(personList.getItems().get(selectedIndex));
+                                                    new PersonDeletedStatus(tmpPerson));
+            personList.getItems().get(selectedIndex).setIsDeleted(true);
+            requestExecutor.schedule(()
+                    -> Platform.runLater(() -> modelManager.deletePerson(tmpPerson)), 3, TimeUnit.SECONDS);
```
Comment (YijinL) [inline src/main/java/address/controller/PersonOverviewController.java]: Platform.runLater returns instantly and queues the task on the UI thread, shouldn't use an executor here

**Pull request merged**